File: source/includes/fts-tutorial/date-number-to-string/autocomplete-number-to-string-query.java
import java.util.Arrays;
import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Aggregates.limit;
import static com.mongodb.client.model.Aggregates.project;
import static com.mongodb.client.model.Projections.excludeId;
import static com.mongodb.client.model.Projections.fields;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class DateNumberToStringQuery {
	public static void main( String[] args ) {
		// define query
		Document agg = new Document("$search",
		new Document ("index", "date-number-fields-tutorial")
			.append("compound",
				new Document("should", Arrays.asList(
					new Document("autocomplete", 
						new Document("path", "maximumNumberOfNights")
						.append("query", "3")), 
					new Document("autocomplete", 
						new Document("path", "accommodatesNumber")
						.append("query", "2"))))));

		// specify connection
		String uri = "<connection-string>";
		
    	// establish connection and set namespace
		try (MongoClient mongoClient = MongoClients.create(uri)) {
			MongoDatabase database = mongoClient.getDatabase("sample_airbnb");
			MongoCollection<Document> collection = database.getCollection("airbnb_mat_view");
			// run query and print results
			collection.aggregate(Arrays.asList(
					eq("$search", eq("compound", agg)), 
					limit(5), 
					project(fields(excludeId()) ))
			).forEach(doc -> System.out.println(doc.toJson()));	
		}
	}
}

File: content/search/source/includes/string-tutorial/code-snippets/go/autocomplete-number-to-string-query.go
package main

import (
	"context"
	"fmt"
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
	"go.mongodb.org/mongo-driver/v2/mongo/options"
)

// define structure of movies collection
type MovieCollection struct {
	title string `bson:"Title,omitempty"`
}

func main() {
	var err error
	// connect to the Atlas cluster and set a maximum operation time
	ctx := context.Background()
	opts := options.Client().
		SetTimeout(5 * time.Second).
		ApplyURI("<connection-string>")

	client, err := mongo.Connect(opts)
	if err != nil {
		panic(err)
	}
	defer client.Disconnect(ctx)
	// set namespace
	collection := client.Database("sample_airbnb").Collection("airbnb_mat_view")
	// define pipeline
	searchStage := bson.D{{"$search", bson.M{
		"index": "date-number-fields-tutorial",
		"compound": bson.M{
			"should": bson.A{
				bson.M{
					"autocomplete": bson.M{
						"path": "maximumNumberOfNights", "query": "3",
					},
				},
				bson.M{
					"autocomplete": bson.M{
						"path": "accommodatesNumber", "query": "2",
					},
				},
			},
		},
	}}}
	limitStage := bson.D{{"$limit", 5}}
	projectStage := bson.D{{"$project", bson.D{{"_id", 0}}}}
	// run pipeline
	cursor, err := collection.Aggregate(ctx, mongo.Pipeline{searchStage, limitStage, projectStage})
	if err != nil {
		panic(err)
	}
	// print results
	var results []bson.D
	if err = cursor.All(context.TODO(), &results); err != nil {
		panic(err)
	}
	for _, result := range results {
		fmt.Println(result)
	}
}
